### PR TITLE
Add `ignore_hash` option in `settings.ini`

### PR DESCRIPTION
**Pull request by lebarsfa**
Based on the work of @mardy and as a quick workaround for some problems described in #521 and #224, here is a proposition to add `ignore_hash` option in `settings.ini`. When set to `True`, `sha256` hashes would not be checked.

### CODE REVIEW

**Comment by ddalcino** [comment]
You could save yourself and a lot of users some trouble by checking the sha1 hash (just pass `sha1` to the get_hash function), rather than not checking the sha256 hash at all. The sha1 hash is pretty much always available at the mirrors. Failure to check the hash at all makes it pretty easy to download corrupted files.

We started having trouble with #521 when we moved to sha256 for security reasons. The sha256 hash is often unavailable at mirrors, and not available for a few hours after new releases are uploaded.

Whatever you call the new option, I think we need to make it extremely clear that this option is insecure and not for production use. Personally, I would include the text `INSECURE_NOT_FOR_PRODUCTION` in the name of the option itself. We don't want to see malware getting into anyone's production binaries because someone didn't understand the consequences of turning on this option.

Also, this feature is going to need documentation (see the `docs` folder).

**Comment by lebarsfa** [comment]
Here are some changes:
- As suggested, renamed `ignore_hash` option to `INSECURE_NOT_FOR_PRODUCTION_ignore_hash`.
- New `hash_algorithm` option, to be able to choose `sha1` instead of `sha256`. However, if https://download.qt.io/ is not available, it seems the desired mirror will need to be added to `trusted_mirrors` list for the hashes to be downloaded from there. Unfortunately, it appears that the mirrors often do not provide any hash for Updates.xml, so I proposed to just show a warning for that specific file when this happens, and continue the downloads of the other files, which should have available hashes.
- Documentation for both options.

I am not sure of what to do with sonatype-lift comments, I did not have them on my fork...

File: aqt/installer.py [hunk 1]
```diff
@@ -1209,12 +1209,12 @@ def installer(
...
     #
     timeout = (Settings.connection_timeout, Settings.response_timeout)
-    hash = get_hash(qt_package.archive_path, algorithm="sha256", timeout=timeout)
+    hash = get_hash(qt_package.archive_path, Settings.hash_algorithm, timeout) if not Settings.ignore_hash else None
 
     def download_bin(_base_url):
         url = posixpath.join(_base_url, qt_package.archive_path)
         logger.debug("Download URL: {}".format(url))
-        return downloadBinaryFile(url, archive, "sha256", hash, timeout)
+        return downloadBinaryFile(url, archive, Settings.hash_algorithm, hash, timeout)
```
[the remark this answers is not in the document]
Comment (miurahr) [reply aqt/installer.py:1217]: As Sonatype pointed out for you @lebarsfa , `downloadBinaryFile` method only accept argument passed by `hash` should be `bytes` variable type.

see `def downloadBinaryFile(url: str, out: Path, hash_algo: str, exp: bytes, timeout: Tuple[float, float]) -> None:`

You make variable `hash` able to be None

`   hash = get_hash(qt_package.archive_path, Settings.hash_algorithm, timeout) if not Settings.ignore_hash else None`

Please fix a type inconsistency.

File: aqt/helper.py [hunk 2]
```diff
@@ -94,7 +94,14 @@ def getUrl(url: str, timeout: Tuple[float, float], expected_hash: Optional[bytes
                 raise ArchiveDownloadError(msg)
         result: str = r.text
         filename = url.split("/")[-1]
-        actual_hash = hashlib.sha256(bytes(result, "utf-8")).digest()
+        if Settings.hash_algorithm == "sha256":
+            actual_hash = hashlib.sha256(bytes(result, "utf-8")).digest()
+        elif Settings.hash_algorithm == "sha1":
+            actual_hash = hashlib.sha1(bytes(result, "utf-8")).digest()
+        elif Settings.hash_algorithm == "md5":
+            actual_hash = hashlib.md5(bytes(result, "utf-8")).digest()
```
[the remark this answers is not in the document]
Comment (miurahr) [reply aqt/helper.py:102]: As Sonatype AI bot review your code and pointed out, we should not use "sha1" and "md5" for security.
Your change may lead a misunderstandings of users to be able to use these for security and protect users, but not.

Please read the python manual to know what is `useforsecurity=False` 
https://docs.python.org/3/library/hashlib.html#constructors

**Review by miurahr — CHANGES_REQUESTED** [review]
Please fix the review comments that Sonatype bot give you

File: aqt/helper.py [hunk 3]
```diff
@@ -94,7 +94,14 @@ def getUrl(url: str, timeout: Tuple[float, float], expected_hash: Optional[bytes
                 raise ArchiveDownloadError(msg)
         result: str = r.text
         filename = url.split("/")[-1]
-        actual_hash = hashlib.sha256(bytes(result, "utf-8")).digest()
+        if Settings.hash_algorithm == "sha256":
+            actual_hash = hashlib.sha256(bytes(result, "utf-8")).digest()
+        elif Settings.hash_algorithm == "sha1":
+            actual_hash = hashlib.sha1(bytes(result, "utf-8")).digest()
```
[the remark this answers is not in the document]
Comment (miurahr) [reply aqt/helper.py]: @sonotype-lift ignore
We use sha1 as an alternative of sha256 and not for strong security and give warning in documentation.

File: aqt/helper.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (miurahr) [reply aqt/helper.py]: applied and pushed to your branch.

**Review by miurahr — APPROVED** [review]

**Pull request merged**